frontend/catalyst/compiler.py

```diff
@@ -155,6 +156,55 @@ def run_writing_command(command: List[str], compile_options: Optional[CompileOpt
         "adjoint-lowering",
     ],
 )
+QUANTUM_COMPILATION_PASS_deafult_len = len(QUANTUM_COMPILATION_PASS[1])
+
+"""
+# We would like the quantum passes pipeline to look something like the following:
+QUANTUM_COMPILATION_PASS = (
+    "QuantumCompilationPass",
+    [
+        # Begin quantum peephole
+        "func.func(remove-chained-self-inverse{func-name=circuit})",
+        "func.func(remove-chained-self-inverse{func-name=toy})",
+        ...
+        # End quantum peephole
+        "annotate-function",
+        ...
+    ],
+)
+
+The peephole circuit transforms should occur before all the lowering
+because the circuit transforms are independent of the lowering and do 
+not need to parse through all the lowered MLIR operations/instructions. 
+"""
+
+
+def fill_quantum_peephole_passes_pipeline(pass_table):
```

Comment:
TODO: processing the pipeline with the table is no longer necessary since we switched to the transform dialect. The only line needed here is `"transform-interpreter"`.